### PR TITLE
Require default for finite domains instead of giving up

### LINKED ISSUE #652: Overzealous exhaustiveness checker for booleans
The following (IMO valid) program
```scala
def foo(opt: Option[Bool]) = opt match {
  case Some(false) => Some(false)
  case opt => Some(true)
}
```
returns an error `Non-exhaustive pattern matching, missing case Some(true)` which is clearly nonsense here.

Using the `else` for a `match` also doesn't work
```scala
def foo(opt: Option[Bool]) = opt match {
  case Some(false) => Some(false)
} else Some(true)
```

---

Moreover, even the following program (and its `else` variant) fails with the `Non-exhaustive pattern matching, missing case true` error:
```scala
def foo(b: Bool) = b match {
  case false => false
  case b => b
}
```

### PR DESCRIPTION
**Pull request by b-studios**
Should fix #652

### CODE REVIEW

File: examples/pos/issue652.effek
Comment (jiribenes) [inline examples/pos/issue652.effek:1]: oops 😇

**Comment by jiribenes** [comment]
It seems that it might be breaking my nice custom error messages in `neg/match_nested_literal.effekt` 🥲 

```
==> X effekt.JavaScriptTests.examples/ (js)  0.04s munit.ComparisonFailException: 
=> Diff (- obtained, + expected)
-[error] examples/neg/match_nested_literal.effekt:9:21: Non-exhaustive pattern matching, scrutinees of type Bool require a default case Or(_, And(Lit(_), _))
+[error] examples/neg/match_nested_literal.effekt:9:21: Non-exhaustive pattern matching, missing case Or(_, And(Lit(true), _))
 def test(b: BExp) = b match {
                     ^
-[error] examples/neg/match_nested_literal.effekt:9:21: Non-exhaustive pattern matching, scrutinees of type Bool require a default case Lit(_)
+[error] examples/neg/match_nested_literal.effekt:9:21: Non-exhaustive pattern matching, missing case Lit(false)
```

File: examples/pos/issue652.effek
Comment (jiribenes) [reply examples/pos/issue652.effek:1]: @b-studios 👀 You still left this file here

Comment (b-studios) [reply examples/pos/issue652.effek:1]: haha, oh that's true

**Pull request merged**